quickstarts/microsoft.devcenter/environment-sandbox/main.bicep

```diff
@@ -0,0 +1,87 @@
+targetScope = 'subscription'
+
+@description('Name of the resource group')
+param resourceGroupName string
+
+@description('Location to create resources in')
+@allowed(
+  [
+    'australiaeast'
+    'brazilsouth'
+    'canadacentral'
+    'centralindia'
+    'centraluseuap'
+    'eastasia'
+    'eastus'
+    'eastus2'
+    'japaneast'
+    'koreacentral'
+    'northeurope'
+    'southafricanorth'
+    'southcentralus'
+    'southeastasia'
+    'switzerlandnorth'
+    'uksouth'
+    'westeurope'
+    'westus2'
+    'westus3'
+  ]
+)
+param resourceLocation string = 'westus3'
+
+@description('Principal/object ID of the user to assign role assignments to')
+param principalId string
+
+@description('Name of Dev Center')
+@minLength(3)
+@maxLength(26)
+param devcenterName string
+
+@description('Name of Project associated with Dev Center')
+@minLength(3)
+@maxLength(63)
+param projectName string
+
+@description('Name of Environment Type e.g. Sandbox, Dev, Prod')
+@minLength(3)
+@maxLength(63)
+param environmentTypeName string = 'Sandbox'
+
+@description('GUID used as name of role assignment')
+param guidSeed string = newGuid()
+
+var devcenterResourceId = resourceId(subscription().subscriptionId, resourceGroupName, 'Microsoft.DevCenter/devcenters', devcenterName)
```

Comment:
this is a nitpick, but it would be more idiomatic to use an `existing` resource declaration to retrieve the ID which can validate more deeply that generating your own resource ID:

```bicep
resource devCenter 'microsoft.devcenter/devcenters@<date>' existing = {
  name: devcenterName
  scope: resourceGroup(resourceGroupName)
}

// ...

module deployment2 './modules/deployment2.bicep' = {
  scope: subscription(subscription().subscriptionId)
  name: guid('Deployment 2', guidSeed)
  ...
  params: {
    devcenterResourceId: devCenter.id
  }
}
```

Reply:
Ack. Updated in latest commit to use existing resource ID